### PR TITLE
Rubocop fixes

### PR DESCRIPTION
**Pull request by ineverov**
Fixed line length offenses for `lib/` (except `element_container.rb` -  there are several PR changing this file)

### CODE REVIEW

File: lib/site_prism/addressable_url_matcher.rb [hunk 1]
```diff
@@ -72,29 +72,33 @@ def extract_component_templates
...
+    # Returns empty hash if the template omits the component,
+    # a set of substitutions if the
+    # provided URI component matches the template component,
+    # or nil if the match fails.
     def component_matches(component, uri)
-      extracted_mappings = {}
-      component_template = component_templates[component]
-      if component_template
+      mappings = {}
+      if (component_template = component_templates[component])
```
Comment (luke-hill) [inline lib/site_prism/addressable_url_matcher.rb]: I personally dislike this kind of assignment, so I altered it to the previous one

File: lib/site_prism/addressable_url_matcher.rb [hunk 2]
```diff
@@ -72,29 +72,33 @@ def extract_component_templates
...
         component_url = uri.public_send(component).to_s
-        extracted_mappings = component_template.extract(component_url)
-        unless extracted_mappings
+        mappings = component_template.extract(component_url)
+        unless mappings
           # to support Addressable's expansion of queries
           # ensure it's parsing the fragment as appropriate (e.g. {?params*})
           prefix = COMPONENT_PREFIXES[component]
-          return nil unless prefix && (extracted_mappings = component_template.extract(prefix + component_url))
+          return nil unless prefix
```
Comment (luke-hill) [inline lib/site_prism/addressable_url_matcher.rb]: Is this line the same (And some weird ruby-ness I don't know)

File: lib/site_prism/addressable_url_matcher.rb [hunk 3]
```diff
@@ -115,28 +119,32 @@ def substitutions
...
 
     def reverse_substitutions
-      @reverse_substitutions ||= slugs.each_with_index.reduce({}) do |memo, slug_index|
-        slug, index = slug_index
-        memo.merge(slug_prefix(slug) + substitution_value(index) => slug, substitution_value(index) => slug)
-      end
+      @reverse_substitutions ||=
+        slugs.each_with_index.reduce({}) do |memo, slug_index|
+          slug, index = slug_index
+          memo.merge(slug_prefix(slug) + substitution_value(index) => slug,
```
Comment (luke-hill) [inline lib/site_prism/addressable_url_matcher.rb]: if we're expanding the brackets could we go down the
```
method(
  stuff
)
```

route

File: lib/site_prism/exceptions.rb [hunk 4]
```diff
@@ -6,7 +6,8 @@ class NoUrlMatcherForPage < StandardError; end
 
   class InvalidUrlMatcher < StandardError
     def message
-      'Could not automatically match your URL. Templated port numbers are unsupported.'
+      "Could not automatically match your URL. \
+Templated port numbers are unsupported."
```
Comment (luke-hill) [inline lib/site_prism/exceptions.rb]: this needs indenting once from the previous line

File: lib/site_prism/loadable.rb [hunk 5]
```diff
@@ -35,24 +39,30 @@ def self.included(base)
...
+      previously_loaded = loaded
+      message = 'A block was expected, but none received.'
+      raise ArgumentError, message unless block_given?
 
       # Within the block, cache loaded? to optimize performance.
       self.loaded = loaded?
       unless loaded
-        message = "Failed to load because: #{load_error || 'no reason given'}"
-        raise(::SitePrism::NotLoadedError, message)
+        raise(::SitePrism::NotLoadedError,
```
Comment (luke-hill) [inline lib/site_prism/loadable.rb]: lets keep this one as per previous to keep the consistency

File: lib/site_prism/addressable_url_matcher.rb (hunk 1 above)
Comment (ineverov) [reply lib/site_prism/addressable_url_matcher.rb]: I was trying to leverage between line length and method length. Currently, we have 80 chars in 11 (code) lines

File: lib/site_prism/addressable_url_matcher.rb (hunk 2 above)
Comment (ineverov) [reply lib/site_prism/addressable_url_matcher.rb]: original condition `return nil if prefix is nil` or `component_template.extract(prefix + component_url) is nil`. In the first case we can immediately return nil (new line 93). in second case mapping will be nil and we return nil at the end of the method (line 97)

File: lib/site_prism/addressable_url_matcher.rb (hunk 3 above)
Comment (ineverov) [reply lib/site_prism/addressable_url_matcher.rb]: yes, sure we can, but again we have cop to check method length < 11 lines

File: lib/site_prism/exceptions.rb (hunk 4 above)
Comment (ineverov) [reply lib/site_prism/exceptions.rb]: in this case, you'll have extra spaces in the message.
Now,` "Could not automatically match your URL. Templated port numbers are unsupported."`
if ident `"Could not automatically match your URL.          Templated port numbers are unsupported."`

File: lib/site_prism/loadable.rb (hunk 5 above)
Comment (ineverov) [reply lib/site_prism/loadable.rb]: again leverage between line and method length... :(

File: lib/site_prism/addressable_url_matcher.rb (hunk 2 above)
Comment (luke-hill) [reply lib/site_prism/addressable_url_matcher.rb]: Yes... we can... Top spot!

File: lib/site_prism/addressable_url_matcher.rb (hunk 3 above)
Comment (luke-hill) [reply lib/site_prism/addressable_url_matcher.rb]: Shouldn't be affected.

**Comment by luke-hill** [comment]
As we've cut the 2.13 gem release can we get this one rebased and fixed up (As it's a simple fix and I just wanna get this in) @ineverov

**Comment by ineverov** [comment]
@luke-hill rebased

File: lib/site_prism/addressable_url_matcher.rb (hunk 1 above)
Comment (luke-hill) [reply lib/site_prism/addressable_url_matcher.rb]: Can we revert this one?

File: lib/site_prism/addressable_url_matcher.rb (hunk 3 above)
Comment (luke-hill) [reply lib/site_prism/addressable_url_matcher.rb]: And alert this one as well

File: lib/site_prism/loadable.rb (hunk 5 above)
Comment (luke-hill) [reply lib/site_prism/loadable.rb]: Can we revert this one

File: lib/site_prism/page.rb [hunk 6]
```diff
@@ -10,7 +10,8 @@ class Page
     extend ElementContainer
 
     load_validation do
-      [displayed?, "Expected #{current_url} to match #{url_matcher} but it did not."]
+      [displayed?,
```
Comment (luke-hill) [inline lib/site_prism/page.rb]: for Arrays I think having
```
[
  val1,
  val2,
]
```

Looks the cleanest

**Comment by ineverov** [comment]
@luke-hill Fixed comments. Going to work on element container soon

File: lib/site_prism/page.rb [hunk 7]
```diff
@@ -10,7 +10,10 @@ class Page
     extend ElementContainer
 
     load_validation do
-      [displayed?, "Expected #{current_url} to match #{url_matcher} but it did not."]
+      [
```
Comment (ineverov) [inline lib/site_prism/page.rb]: This change made Page class longer than 100 lines. This is the reason rubocop is failing

**Pull request merged**